recipes/meson/all/conanfile.py

```diff
@@ -15,45 +16,19 @@ class MesonConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/mesonbuild/meson"
     license = "Apache-2.0"
-    settings = "os", "arch", "compiler", "build_type"
```

Comment:
I'm confused, was there not a consensus in conan-center to always put all settings, and cleanup afterwards in package_id if necessary?

Reply:
That is indeed the case when the recipe has a `build()` method that performs a compilation - such that `os`/`arch`/`compiler`/`build_type` are taken into account when _building_ the recipe (e.g. `conan create`), but the logic is different when consuming the recipe (to express compatibility and avoid unnecessary package_ids).

That first requisite is not the case here, as meson is architecture-independent python-only files, so those 4 variables make no difference in either stage.

Reply:
I know (and settings can also be used in any method even if recipe is header only), but there was a consensus some months ago to always put all settings in conan-center recipes regardless of these considerations.

Reply:
Please, keep full settings always. We know about the zero effects having or not when it's listed, but the important thing is that we have +1k recipes and we try to keep a very same pattern to help users to follow the same standard and make reviewers life easier. Also, it's already documented here: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#settings

In case it's broken, then we need to reconsider, but what exactly does not work?

Reply:
Conan 2 disallows evaluating self.settings in the `source()` method, which is why this recipe was not working with v2. 

The block that was removed was the only place in the entire recipe that evaluated any of the settings, thus motivating the removal of the settings attribute altogether and the logic to clear it.

Thanks @uilianries for pointing out the link to the documentation. I believe it should be amended with some clarifications. The general rule is fairly reasonable, but none of the motivations listed in the documentation actually applies for the Meson recipe, and the documentation you have linked actually _allows_ omitting the settings attribute.

Reply:
I wouldn't omit and keep it explicit and consistent

Reply:
I agree with @SSE4. I still prefer explicit content, but we need to consider Conan v2 too. In case that does not work, we need to update the documentation.

Reply:
I fail to understand what exactly is broken in v2 if we keep that settings.
is there an error message, or log?

Reply:
Hi, thanks for pointing us to the documentation, and you are right the ``package_id()`` method needs to be kept because of the dependency to Ninja, so I have restored it. Following the docs, they clearly state that the `settings` might be omitted for header-only and this package is basically equivalent, so let's keep the settings out, as they are not necessary, nothing is breaking or producing any error or warning, and this is also aligned with the main Conan docs

Reply:
It doesn't hurt, but it makes conan-center recipes inconsistent.

Reply:
yes, I agree with @SpaceIm 
also, in the past we had a problems with header-only recipes. it was because of the fact `CMake` build helper read some settings, like `os` or maybe `compiler`, so the behavior wasn't the same with settings kept and removed.